corehq/apps/users/views/mobile/users.py

```diff
@@ -1129,7 +1132,55 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
-class DeleteCommCareUsers(BaseManageCommCareUserView):
+class UsernameUploadMixin(object):
+    """
+    Contains helper functions for working with a file that consists of a single column of usernames.
+    """
+
+    def _get_usernames(self, request):
+        """
+            Get username list from Excel supplied in request.FILES.
+            Adds any errors to request.messages.
+        """
+        try:
+            workbook = get_workbook(request.FILES.get('bulk_upload_file'))
+        except WorkbookJSONError as e:
+            messages.error(request, str(e))
+            return None
+
+        try:
+            sheet = workbook.get_worksheet()
+        except WorksheetNotFound:
+            messages.error(request, _("Workbook has no worksheets"))
+            return None
```

Comment:
might be worth extracting _validate_workbook to put these two in there.

Reply:
d41b116cac8